app/connectors/StatusCheckResponseHttpParser.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import uk.gov.hmrc.http.{HttpReads, HttpResponse}
+import play.api.http.Status._
+import play.api.Logging
+import scala.util.{Failure, Success, Try}
+import models.{StatusCheckError, StatusCheckErrorResponse, StatusCheckErrorResponseWithStatus, StatusCheckResponse}
+import connectors.ErrorCodes._
+
+object StatusCheckResponseHttpParser extends Logging {
+
+  implicit object StatusCheckResponseReads
+      extends HttpReads[Either[StatusCheckErrorResponseWithStatus, StatusCheckResponse]] {
+
+    val HEADER_X_CORRELATION_ID = "X-Correlation-Id"
```

Comment:
scratch making the other one private can we make this global so if it ever changes it changes in one place